### DISCUSSION TITLE
Problema test_register cuando admin crea un usuario

### DISCUSSION THREAD

**Discussion by MarioArocaPaez** [MarioArocaPaez/decide#9] [Q&A]
Como ya sabéis, estoy trabajando en la implementación del registro para usuarios normales y, en principio, todo va bien. Si probáis el código que tengo subido en mi rama, veréis que un usuario puede registrarse por sí mismo, y también el administrador puede seguir registrando usuarios sin problemas. La cuestión es que cuando ejecuto los tests del módulo de autenticación, el test **test_register()** me da fallo, ya que el test espera un código de respuesta **201**, pero no sé por qué recibe un **400**.

A pesar de lo que dice el test, si intentamos crear un usuario siendo administrador, este se crea sin problemas, y además, si observamos la red, no se registra ningún código de respuesta 400. :skull:

Este es el error que el test me está arrojando. A ver si se os ocurre algo :sweat_smile: (Quizás tiene algo que ver con el 302 que me da al hacer el post no sé)

`......F..
======================================================================
FAIL: test_register (authentication.tests.AuthTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mario/egc/decide/decide/authentication/tests.py", line 116, in test_register
    self.assertEqual(response.status_code, 201)
AssertionError: 400 != 201

----------------------------------------------------------------------
Ran 9 tests in 2.391s

`
![image](https://github.com/3ralon/decide/assets/73229438/005078a3-d168-4cb1-9546-a27438c53a85)

**Comment by laurolmer** [comment]
Investigando el problema, me he dado cuenta de que a la hora de hacer los tests del registro de usuarios, no tienes en cuenta la segunda contraseña de verificación, solamente pones el usuario y la primera contraseña.

**Reply by MarioArocaPaez** [reply]
Era justo eso, ya tiran bien los tests. Muchas gracias profe :angel:

**Comment by MarioArocaPaez** [comment]
En efecto era lo que Laura comentaba, copié los test para el login y esos solo se esperan un user y password. Sin embargo para register hace falta user, password1 y password2
